packages/template-retail-react-app/app/hooks/use-password-reset.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {AuthHelpers, useAuthHelper} from '@salesforce/commerce-sdk-react'
+import {useToast} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {useIntl} from 'react-intl'
+import {absoluteUrl} from '@salesforce/retail-react-app/app/utils/url'
+/**
+ * This hook provides commerce-react-sdk hooks to simplify the reset password flow.
+ */
+export const usePasswordReset = () => {
+    const toast = useToast()
+    const {formatMessage} = useIntl()
+
+    const getPasswordResetTokenMutation = useAuthHelper(AuthHelpers.GetPasswordResetToken)
+    const resetPasswordMutation = useAuthHelper(AuthHelpers.ResetPassword)
+
+    const getPasswordResetToken = async (email) => {
+        await getPasswordResetTokenMutation.mutateAsync({
+            user_id: email,
+            callback_uri: absoluteUrl('/reset-password-callback')
+        })
+    }
+
+    const resetPassword = async ({email, token, newPassword}) => {
+        await resetPasswordMutation.mutateAsync(
+            {user_id: email, pwd_action_token: token, new_password: newPassword},
+            {
+                onSuccess: () => {
+                    toast({
+                        title: formatMessage({
+                            defaultMessage: 'Password Reset Success',
+                            id: 'password_reset_success.toast'
+                        }),
+                        status: 'success',
+                        position: 'bottom-right',
+                        isClosable: true
+                    })
+                }
```

Comment:
How are we handling error here?

Reply:
I handled error states in the submit handlers:

getResetPasswordToken: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2132/files/ad499f5973eeb37f211c857ba214357345933f2b#diff-f75f0fcf60f7cddf7b77788a61e94b5db98a2a48dd0ce2c079bd877f956952c3R174-R177
resetPassword: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2132/files/ad499f5973eeb37f211c857ba214357345933f2b#diff-e7360d13aeb062a7fad4c0655472bd2d1e2b7f82f5e8f444d6fc281f9f0d3316R49-R52

Reply:
We also have a separate [ticket](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000026ueL6YAI/view) in the next sprint to go through all of the error states and make sure they are covered with appropriate and clear error handling and messages!